subgraphs/compound-forks/src/mapping.ts

```diff
@@ -1640,6 +1870,217 @@ function getMarketDailySnapshotID(marketID: string, timestamp: i32): string {
   return marketID.concat("-").concat((timestamp / SECONDS_PER_DAY).toString());
 }
 
+// A series of side effects on position added
+// They include:
+// * Create a new position when needed or reuse the exisitng position
+// * Update position related data in protocol, market, account
+// * Take position snapshot
+function addPosition(
+  protocol: LendingProtocol,
+  market: Market,
+  account: Account,
+  balance: BigInt,
+  side: string,
+  eventType: EventType,
+  event: ethereum.Event
+): string {
+  let counterID = account.id
+    .concat("-")
+    .concat(market.id)
+    .concat("-")
+    .concat(side);
+  let positionCounter = _PositionCounter.load(counterID);
+  if (!positionCounter) {
+    positionCounter = new _PositionCounter(counterID);
+    positionCounter.nextCount = 0;
+    positionCounter.save();
+  }
+  let positionID = positionCounter.id
+    .concat("-")
+    .concat(positionCounter.nextCount.toString());
+
+  let position = Position.load(positionID);
+  let openPosition = position == null;
+  if (openPosition) {
+    position = new Position(positionID);
+    position.account = account.id;
+    position.market = market.id;
+    position.hashOpened = event.transaction.hash.toHexString();
+    position.blockNumberOpened = event.block.number;
+    position.timestampOpened = event.block.timestamp;
+    position.side = side;
+    if (side == PositionSide.LENDER) {
+      position.isCollateral =
+        account._enabledCollaterals.indexOf(market.id) >= 0;
+    }
+    position.balance = BIGINT_ZERO;
+    position.depositCount = 0;
+    position.withdrawCount = 0;
+    position.borrowCount = 0;
+    position.repayCount = 0;
+    position.liquidationCount = 0;
+    position.save();
+  }
+  position = position!;
+  if (balance.ge(BIGINT_ZERO)) {
```

Comment:
I think it'll be easier to understand if you test for equality with the named constant